### DISCUSSION TITLE
If its possible to plot "edge to edge"?

### DISCUSSION THREAD

**Discussion by Oaklight** [networkx/networkx#5997] [Q&A] [answered]
I'm using networkx to analyze Structural model equation graphs from some papers. There are cases where some nodes are called "moderators". Essentially it's a node (A) pointing to some edge (E) and it means such node A could tune the correlation btw nodes connected by E.
Here is an example:
![image](https://user-images.githubusercontent.com/13076552/191808833-10a30fbe-d2a6-4ce5-b531-ca667cda10b0.png)

Employee Involvement Climate is a moderator to edge H2
Now, I'm **trying to plot H8a/H8b** connection in the above graph. Is there a way to use networkx to accomplish this?

Here is what I got without the moderator link:
![image](https://user-images.githubusercontent.com/13076552/191809262-4fb31d3f-2105-4da9-aafb-b5aaf9833f0b.png)

**Comment by rossbar** [comment, marked as answer]
AFAIK there is no native way to draw edges that terminate on edges - the edge drawing always terminates at a node. However; this means there's a straightforward way to implement this by creating a node on the edge that you want to target. A super quick example:

```python
>>> import networkx as nx
>>> G = nx.DiGraph()
>>> G.add_nodes_from(range(3))
# Create an edge
>>> G.add_edge(0, 1)
>>> pos = {0: (0, 0), 1: (1, 0), 2: (0.5, 1)}
# Draw the existing edge
>>> nx.draw(G, pos=pos, with_labels=True)
# Add & position a "phantom" node at the center of the edge
>>> G.add_node((0, 1))
>>> pos[(0, 1)] = (0.5, 0)
>>> G.add_edge(2, (0, 1))
>>> nx.draw_networkx_edges(G, pos, edgelist=[(2, (0, 1))], node_size=0)
```

Which gives:

![edge2edge](https://user-images.githubusercontent.com/1268991/191816030-3051ecbb-3f8e-44a0-9c44-6129e1c21be2.png)

In this basic example the node positions are determined manually, in practice you'd probably want to determine the "phantom" node positions programmatically, which should be straightforward using the layout dict.

**Reply by Oaklight** [reply]
I realize it could be faked by an additional node. I have hundreds of such graphs, and each of them is plotted using program. I don't know the pos of the middle point of 0-1 edge ahead of time. Is there a way to get such middle node pos info?

**Reply by Oaklight** [reply]
okay, the pos of nodes in edge could be returned by circular_layout, then middle node pos could be calculated and then just add another node without plotting it. then plot the link separately.
